contracts/lib/BasicOrderFulfiller.sol

```diff
@@ -381,30 +408,67 @@ contract BasicOrderFulfiller is OrderValidator {
         // Ensure this function cannot be triggered during a reentrant call.
         _setReentrancyGuard(false); // Native tokens rejected during execution.
 
-        // Ensure current timestamp falls between order start time and end time.
-        _verifyTime(parameters.startTime, parameters.endTime, true);
-
         // Verify that calldata offsets for all dynamic types were produced by
         // default encoding. This ensures that the constants used for calldata
         // pointers to dynamic types are the same as those calculated by
         // Solidity using their offsets. Also verify that the basic order type
         // is within range.
         _assertValidBasicOrderParameters();
 
-        {
-            // Retrieve total number of additional recipients & place on stack.
-            uint256 totalAdditionalRecipients;
-            assembly {
-                totalAdditionalRecipients := calldataload(
-                    BasicOrder_additionalRecipients_length_cdPtr
+        // Check for invalid time and missing original consideration items.
+        // Utilize assembly so that constant calldata pointers can be applied.
+        assembly {
+            // Ensure current timestamp is between order start time & end time.
+            if iszero(
+                and(
+                    iszero(
+                        gt(
+                            calldataload(BasicOrder_startTime_cdPtr),
+                            timestamp()
+                        )
+                    ),
+                    gt(calldataload(BasicOrder_endTime_cdPtr), timestamp())
+                )
+            ) {
```

Comment:
can be simplified to:

```solidity
if or(
  gt(
    calldataload(BasicOrder_startTime_cdPtr),
    timestamp()
  ),
  iszero(
    gt(calldataload(BasicOrder_endTime_cdPtr), timestamp())
  )
) { ... }
```

Reply:
If you are willing to make a small deviation in how `endTime` is defined, then 
```solidity
iszero(
    gt(calldataload(BasicOrder_endTime_cdPtr), timestamp())
  )
```
Can be replaced by
```solidity
    lt(calldataload(BasicOrder_endTime_cdPtr), timestamp())
```

The difference is how `endTime == block.timestamp` is handled :)